core/tests/puppeteer-acceptance-tests/spec/voiceover-admin-tests/add-voiceover-artist-to-an-exploration.spec.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility functions for voiceover admin page if voiceover admin
+ * can add voiceover artist to an exploration
+ */
+
+import {UserFactory} from '../../puppeteer-testing-utilities/user-factory';
+import {VoiceoverAdmin} from '../../user-utilities/voiceover-admin-utils';
+import testConstants from '../../puppeteer-testing-utilities/test-constants';
+
+const DEFAULT_SPEC_TIMEOUT = testConstants.DEFAULT_SPEC_TIMEOUT;
+const ROLES = testConstants.Roles;
+
+describe('Voiceover Admin', function () {
+  let voiceoverAdm: VoiceoverAdmin;
+  let explorationEditor: VoiceoverAdmin;
+  let explorationUrl: string | null;
+
+  beforeAll(async function () {
+    voiceoverAdm = await UserFactory.createNewUser(
+      'voiceoverAdm',
+      'voiceover_admin@example.com',
+      [ROLES.VOICEOVER_ADMIN]
+    );
+
+    explorationEditor = await UserFactory.createNewUser(
+      'explorationEditor',
+      'exploration_creator@example.com'
+    );
+  }, DEFAULT_SPEC_TIMEOUT);
+
+  it(
+    'should be able to see error while adding an invalid user as a voiceover artist to an exploration',
+    async function () {
+      await explorationEditor.navigateToCreatorDashboardPage();
+      await explorationEditor.createExplorationWithTitle('Exploration one');
+      explorationUrl =
+        await explorationEditor.publishExplorationWithTitle('Exploration one');
+
+      await voiceoverAdm.navigateToExplorationEditor(explorationUrl);
+      await voiceoverAdm.navigateToExplorationSettingsTab();
+
+      await voiceoverAdm.expectVoiceoverArtistNotExists('invalidUserId');
+      await voiceoverAdm.addVoiceoverArtistToExploration('invalidUserId');
+
+      await voiceoverAdm.expectToSeeErrorToastMessage(
+        'Sorry, we could not find the specified user.'
+      );
+      await voiceoverAdm.closeToastMessage();
+    },
+    DEFAULT_SPEC_TIMEOUT
+  );
+
+  it(
+    'should be able to add regular user as voiceover artist to an exploration',
+    async function () {
+      await UserFactory.createNewUser(
+        'voiceoverartist',
+        'voiceoverartist@example.com'
+      );
+      await explorationEditor.navigateToCreatorDashboardPage();
+      await explorationEditor.createExplorationWithTitle('Exploration two');
+      explorationUrl =
+        await explorationEditor.publishExplorationWithTitle('Exploration two');
```

Comment:
`explorationUrl` seems to be the same as `explorationId` in the implementation of `publishExplorationWithTitle()` method. So does this only contain the exploration id, or does it contain a full URL?

Regardless, I think it's better for the method to return the exploration id. For top-level tests, this would encourage readers to think more in terms of domain concepts like "explorations", as opposed to more technical details like "URLs".

Reply:
Hey @StephenYu2018, Very sorry for the late reply (I was occupied in writing proposal),

Actually, it was indeed `explorationIdUrl`.

>Regardless, I think it's better for the method to return the exploration id. For top-level tests, this would encourage readers to think more in terms of domain concepts like "explorations", as opposed to more technical details like "URLs".

Sure thing, I've updated the test accordingly PTAL

Just one small note (need a little help).
![Screenshot 2024-04-01 at 9 57 37 PM](https://github.com/oppia/oppia/assets/108923011/fe3408cb-2ad0-4571-a0f3-b45419cc6683)

![Screenshot 2024-04-01 at 9 55 46 PM](https://github.com/oppia/oppia/assets/108923011/b921da3a-8053-4b69-a8ca-23e910b3a906)

see this error log that occurred for the first test (`should be able to see an error while adding an invalid user as a voiceover artist to an exploration`), this error is expected and I have written the logic for this expected error. ([here](https://github.com/oppia/oppia/pull/19994/files#diff-484f076a21aa2587eb6f91ef9d38e717fbeb3d1731322623b170b76a2ae481daR198-R213)).

I am not sure if I am missing something here.

Reply:
Hi @hrshkshri,
You can explicitly ignore these using the ConsoleReporter utility (import from console-reporter.ts) and use the setConsoleErrorsToIgnore function to ignore an array of errors using regexes. You can look at the blog-editor-test publish and duplicate spec to see an example. Also try to be explicit e.g. use something like this for the array: [
new RegExp('http:\/\/localhost:8181\/voice_artist_management_handler.*Failed to load resource: the server responded with a status of 400 \(Bad Request\)',
'Sorry, we could not find the specified user.'

Reply:
Hey @jnvtnguyen Thanks, updated the test accordingly.